benchmarks/features.cpp

```diff
@@ -0,0 +1,29 @@
+#include <benchmark/benchmark.h>
+#include <pcl/features/normal_3d.h> // for NormalEstimation
+#include <pcl/io/pcd_io.h> // for PCDReader
+
+static void BM_NormalEstimation(benchmark::State& state, const std::string& file) {
+  // Perform setup here
+  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud (new pcl::PointCloud<pcl::PointXYZ>);
+  pcl::PCDReader reader;
+  reader.read (file, *cloud);
+  pcl::NormalEstimation<pcl::PointXYZ, pcl::Normal> ne;
+  ne.setInputCloud (cloud);
+  ne.setKSearch (100);
+  pcl::PointCloud<pcl::Normal>::Ptr cloud_normals (new pcl::PointCloud<pcl::Normal>);
+  for (auto _ : state) {
+    // This code gets timed
+    ne.compute (*cloud_normals);
```

Comment:
Does this give satisfactory graphs? Or does the compiler optimize the loop to no-op?

Reply:
> Does this give satisfactory graphs? Or does the compiler optimize the loop to no-op?

I'm not sure if this is the same, but the method is only run for one iteration on the CI:
https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=17871&view=logs&jobId=cd6332f9-d180-53f5-8d9a-3b5d9c3252a2&j=cd6332f9-d180-53f5-8d9a-3b5d9c3252a2&t=7e929b77-dd35-56c3-fbaf-db0044c91b15

Reply:
The input file is too large for the benchmark to run it multiple times

Reply:
I just tried with the bun0.pcd and here it runs for about 224 times - so it just seems to be saturated when using 2.5 sec for a benchmark, ie using table_scene_mug_stereo_textured.

Reply:
> The input file is too large for the benchmark to run it multiple times

Saw this after my own test 😄

Reply:
It is not optimized away, I assume that the `for (auto _ : state)` has some hidden do-not-optimize-me-magic.
I think it is possible to request it to run more than once, or for a minimum time, if you think once is not enough.

Reply:
More runs are better since they remove the noise due to ambiguousness of the new cpu architectures (and client sharing on the cloud )

Reply:
That is true. The number of repetitions can be set for each benchmark individually in the code by calling `Repetitions`, or for all benchmarks in one executable with `--benchmark_repetitions=x`. I would suggest the latter, since more flexible (adding it to `ARGUMENTS` of `PCL_ADD_BENCHMARK`)